janusgraph-core/src/main/java/org/janusgraph/graphdb/query/graph/IndexQueryBuilder.java

```diff
@@ -121,8 +121,8 @@ public String getQuery() {
         return query;
     }
 
-    public ImmutableList<Parameter<Order>> getOrders() {
-        return ImmutableList.copyOf(orders);
```

Comment:
There is no need for a new copy of the list because `orders` don't change in `IndexSerializer` where it is accessed. Thus I've changed this code to `Collections.unmodifiableList(orders)` to return a reference to `orders` wrapped in unmodifiableList instead of copying all elements to a new list.